src/main/java/org/disq_bio/disq/HtsjdkReadsRdd.java

```diff
@@ -0,0 +1,38 @@
+package org.disq_bio.disq;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import org.apache.spark.api.java.JavaRDD;
+
+/**
+ * A {@link HtsjdkReadsRdd} is the distributed equivalent of a htsjdk {@link
+ * htsjdk.samtools.SamReader}. It represents a SAM, BAM, or CRAM file stored in a distributed
+ * filesystem, and encapsulates a Spark RDD containing the reads in it.
+ *
+ * <p>Use a {@link HtsjdkReadsRddStorage} to read and write {@link HtsjdkReadsRdd}s.
+ *
+ * @see HtsjdkReadsRddStorage
+ */
+public class HtsjdkReadsRdd {
+
+  private final SAMFileHeader header;
+  private final JavaRDD<SAMRecord> reads;
+
+  public HtsjdkReadsRdd(SAMFileHeader header, JavaRDD<SAMRecord> reads) {
```

Comment:
I wonder if all the public API methods should accept `RDD` instead of `JavaRDD`.  One can always convert to `JavaRDD` where necessary in `impl` classes.

Reply:
The Java Spark APIs use `JavaRDD`, which is why I did this.

Reply:
Right, but ADAM will be calling the public APIs via Scala.  Let me stub that out and see if it makes any difference.

Reply:
Created new issue https://github.com/disq-bio/disq/issues/24 to track further discussion